.github/workflows/ci.yml

```diff
@@ -41,21 +44,84 @@ jobs:
           pre-commit run --all-files
           pytest taf/tests
 
+  build_and_upload_wheel:
+    runs-on: ubuntu-latest
+    needs: run_tests
+    if: github.event_name == 'release'
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          # build it on the minimum version
+          python-version: '3.8'
+
+      - name: Upgrade pip
+        run: |
+          pip install --upgrade pip setuptools wheel
+
+      - name: Install swig
+        run: |
+          sudo apt-get update
+          sudo apt-get install libhdf5-serial-dev zlib1g-dev libatlas-base-dev lcov swig3.0 libpcsclite-dev
+          mkdir -p ~/bin/ && ln -s /usr/bin/swig3.0 ~/bin/swig && export PATH=~/bin/:$PATH
+
+      - name: Install and build TAF
+        run: |
+          pip install .[yubikey]
+          python setup.py sdist bdist_wheel clean --all
+
+      - name: Install publishing dependencies
+        run: |
+          pip install packaging
+          pip install twine==3.8.0
+
+      # - name: Upload to TestPyPi
+      #   run: |
+      #     twine check dist/*
+      #     twine upload --repository-url https://test.pypi.org/legacy/ dist/*
+      #   env:
+      #     TWINE_USERNAME: ${{ secrets.TEST_PYPI_USERNAME }}
+      #     TWINE_PASSWORD: ${{ secrets.TEST_PYPI_PASSWORD }}
+
+      - name: Upload wheels to PyPI
+        run: |
+          twine check dist/*
+          twine upload dist/*
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+
+
   build-and-test-executables:
-    needs: run-tests
+    needs: build_and_upload_wheel
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
     runs-on: ${{ matrix.os }}
-    if: ${{ needs.run-tests.result == 'success' }}
+
     steps:
+
       - name: Checkout Repository
         uses: actions/checkout@v3
 
+      - name: Get Upload URL
+        id: get_upload_url
+        run: echo "${{ github.event.release.upload_url  }}"
+
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
-          python-version: '3.10'
+          # the newest python that we support
+          python-version: '3.12'
```

Comment:
Are there any ways to minimize us forgetting to upgrade this variable once newer python versions are out? Could we get the latest python version from the python-version matrix? Or instead create a new variable `latest_python_version`?

Reply:
I spent a lot of time trying to get the last element of an array. I couldn't not figure out how to do it, if it's even possible. `-1` does not work. Getting the first element of an array is achievable. Probably spent too much time working on it, since it's not even that important to update this version, as long as the version is still supported. Using a variable is not a problem.